core/src/test/java/org/apache/iceberg/actions/TestBinPackRewriteFilePlanner.java

```diff
@@ -455,6 +457,95 @@ void testDeleteRatioThreshold() {
     assertThat(group).containsExactly(tooManyDeletesTask);
   }
 
+  @Test
+  public void testInvalidMaxFilesRewriteParam() {
+    BinPackRewriteFilePlanner planner = new BinPackRewriteFilePlanner(table);
+    Map<String, String> invalidMaxFilesToDeleteOption =
+        ImmutableMap.of(BinPackRewriteFilePlanner.MAX_FILES_TO_REWRITE, "0");
+    assertThatThrownBy(() -> planner.init(invalidMaxFilesToDeleteOption))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot set max-files-to-rewrite to 0, the value must be positive integer.");
+
+    Map<String, String> negativeMaxFilesToDeleteOption =
+        ImmutableMap.of(BinPackRewriteFilePlanner.MAX_FILES_TO_REWRITE, "-2");
+    assertThatThrownBy(() -> planner.init(negativeMaxFilesToDeleteOption))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot set max-files-to-rewrite to -2, the value must be positive integer.");
+  }
+
+  @Test
+  public void textMaxFilesRewriteToOnlyTruncateNeededPartitions() {
+    addFiles();
+    BinPackRewriteFilePlanner planner = new BinPackRewriteFilePlanner(table);
+    Map<String, String> options =
+        ImmutableMap.of(
+            BinPackRewriteFilePlanner.MAX_FILES_TO_REWRITE,
+            "4",
+            BinPackRewriteFilePlanner.REWRITE_ALL,
+            "true");
+    planner.init(options);
+
+    FileRewritePlan<FileGroupInfo, FileScanTask, DataFile, RewriteFileGroup> plan = planner.plan();
+    List<RewriteFileGroup> groups = Lists.newArrayList(plan.groups().iterator());
+
+    List<Integer> fileScanTasksListSize =
+        groups.stream()
+            .map(RewriteGroupBase::fileScanTasks)
+            .map(List::size)
+            .sorted()
+            .collect(Collectors.toList());
+    assertThat(fileScanTasksListSize)
+        .isIn(Arrays.asList(Arrays.asList(1, 3), Arrays.asList(1, 1, 2), Arrays.asList(2, 2)));
+    assertThat(fileScanTasksListSize.stream().reduce(0, Integer::sum)).isEqualTo(4);
+  }
+
+  @Test
+  public void testRewriteMaxFilesOption() {
+    addFiles();
+    BinPackRewriteFilePlanner planner = new BinPackRewriteFilePlanner(table);
+    Map<String, String> options =
+        ImmutableMap.of(
+            BinPackRewriteFilePlanner.MAX_FILES_TO_REWRITE,
+            "5",
+            BinPackRewriteFilePlanner.REWRITE_ALL,
+            "true");
+    planner.init(options);
+
+    FileRewritePlan<FileGroupInfo, FileScanTask, DataFile, RewriteFileGroup> plan = planner.plan();
+    List<RewriteFileGroup> groups = Lists.newArrayList(plan.groups().iterator());
+
+    Integer fileScanTasks =
+        groups.stream()
+            .map(RewriteGroupBase::fileScanTasks)
+            .map(List::size)
+            .reduce(0, Integer::sum);
+    assertThat(fileScanTasks).isEqualTo(5);
+  }
+
+  @Test
+  public void testRewriteMaxFilesOptionInequality() {
```

Comment:
What is the goal of this test?
Is it that, high value don't mess up things?
Maybe `testRewriteMaxFilesOptionNotTriggered`? I'm not sure TBH 😄, and only mention, since the `Integer`->`int` change is needed anyway

Maybe a javadoc comment, just to describe what happens?

Reply:
Exactly !  In case there value provided is still larger than the total number of files , we will make sure that all the files are rewritten . Essentially testing the `max` portion of `max-files-to-rewrite` . I could remove this test if you think this is redundant @pvary  ?

Reply:
I rarely suggest removing tests 😄 
Maybe just rename the test method, or add a comment to describe what this test does.
For me the name was not clear

Reply:
Sure! Thanks!